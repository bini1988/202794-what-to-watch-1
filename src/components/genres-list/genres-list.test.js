import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list";

it(`GenresList correctly renders default markup`, () => {
  const genres = [`Drama`, `Comedy`];
  const tree = renderer
    .create(
        <GenresList
          genres={genres}
          activeGenre="All genres"
          onGenreChange={() => {}}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`GenresList correctly renders empty genres list`, () => {
  const tree = renderer
    .create(
        <GenresList
          genres={[]}
          activeGenre="All genres"
          onGenreChange={() => {}}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
