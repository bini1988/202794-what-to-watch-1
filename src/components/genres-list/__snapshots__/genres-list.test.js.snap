// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList correctly renders default markup 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
</ul>
`;

exports[`GenresList correctly renders empty genres list 1`] = `null`;
